src/rules/no_dupe_class_members.rs

```diff
@@ -0,0 +1,702 @@
+    visitor.visit_module(&module, &module);
+  }
+}
+
+struct NoDupeClassMembersVisitor {
+  context: Context,
+}
+
+impl NoDupeClassMembersVisitor {
+  fn new(context: Context) -> Self {
+    Self { context }
+  }
+
+  fn add_diagnostic(&self, span: Span, name: &str) {
+    self.context.add_diagnostic(
+      span,
+      "no-dupe-class-members",
+      &format!("Duplicate name '{}'", name),
+    );
+  }
+}
+
+impl Visit for NoDupeClassMembersVisitor {
+  fn visit_class(&mut self, class: &Class, parent: &dyn Node) {
+    let mut visitor = ClassVisitor::new(self);
+    visitor.visit_class(class, parent);
+    visitor.aggregate_dupes();
+  }
+}
+
+struct ClassVisitor<'a> {
+  root_visitor: &'a NoDupeClassMembersVisitor,
+  appeared_methods: BTreeMap<MethodToCheck, Vec<(Span, String)>>,
+}
+
+impl<'a> ClassVisitor<'a> {
+  fn new(root_visitor: &'a NoDupeClassMembersVisitor) -> Self {
+    Self {
+      root_visitor,
+      appeared_methods: BTreeMap::new(),
+    }
+  }
+
+  fn aggregate_dupes(&self) {
+    self
+      .appeared_methods
+      .values()
+      .filter(|m| m.len() >= 2)
+      .flatten()
+      .for_each(|(span, name)| {
+        self.root_visitor.add_diagnostic(span.clone(), name);
+      });
+  }
+}
+
+impl<'a> Visit for ClassVisitor<'a> {
+  fn visit_class(&mut self, class: &Class, parent: &dyn Node) {
+    let mut visitor = ClassVisitor::new(self.root_visitor);
+    swc_ecma_visit::visit_class(&mut visitor, class, parent);
+    visitor.aggregate_dupes();
+  }
+
+  fn visit_class_method(
+    &mut self,
+    class_method: &ClassMethod,
+    parent: &dyn Node,
+  ) {
+    if let Some(m) = MethodToCheck::new(
+      &class_method.key,
+      class_method.kind,
+      class_method.is_static,
+    ) {
+      let name = m.normalized_name.clone();
+      self
+        .appeared_methods
+        .entry(m)
+        .or_insert_with(Vec::new)
+        .push((class_method.span, name));
+    }
+    swc_ecma_visit::visit_class_method(self, class_method, parent);
+  }
+}
+
+fn normalize_prop_name(name: &PropName) -> Option<String> {
+  let normalized = match *name {
+    PropName::Ident(Ident { ref sym, .. }) => sym.to_string(),
+    PropName::Str(Str { ref value, .. }) => value.to_string(),
+    PropName::Num(Number { ref value, .. }) => value.to_string(),
+    PropName::Computed(ComputedPropName { ref expr, .. }) => match &**expr {
+      Expr::Lit(Lit::Str(Str { ref value, .. })) => value.to_string(),
+      Expr::Lit(Lit::Bool(Bool { ref value, .. })) => value.to_string(),
+      Expr::Lit(Lit::Null(Null { .. })) => "null".to_string(),
+      Expr::Lit(Lit::Num(Number { ref value, .. })) => value.to_string(),
+      Expr::Lit(Lit::BigInt(BigInt { ref value, .. })) => value.to_string(),
+      Expr::Tpl(Tpl {
+        ref quasis,
+        ref exprs,
+        ..
+      }) if exprs.is_empty() => {
+        quasis.iter().next().map(|q| q.raw.value.to_string())?
+      }
+      _ => return None,
+    },
+  };
```

Comment:
Nice :+1: I could be probably useful in other rules. If so we'll need to move it to other place, but looks good for now 👍